src/Features/CSharp/Portable/CodeFixes/PopulateSwitch/PopulateSwitchCodeFixProvider.cs

```diff
@@ -0,0 +1,196 @@
+                if (containsDefaultLabel)
+                {
+                    // this will not give an index-out-of-bounds error because we know there is at least one case block
+                    newSections = newSections.Insert(switchBlock.Sections.Count - 1, section);
+                }
+                else
+                {
+                    newSections = newSections.Add(section);
+                }
+            }
+            
+            if (!containsDefaultLabel)
+            {
+                newSections = newSections.Add(SyntaxFactory.SwitchSection(SyntaxFactory.List(
+                        new List<SwitchLabelSyntax> { SyntaxFactory.DefaultSwitchLabel() }), statements));
+            }
+
+            var newNode = switchBlock.WithSections(newSections).WithAdditionalAnnotations(Formatter.Annotation, Simplifier.Annotation);
+
+            var newRoot = root.ReplaceNode(switchBlock, newNode);
+            return await Simplifier.ReduceAsync(document.WithSyntaxRoot(newRoot)).ConfigureAwait(false);
+        }
+
+        private static IEnumerable<string> GetMissingLabels(SemanticModel model, List<ExpressionSyntax> caseLabels, INamedTypeSymbol enumType)
+        {
+            var symbols = new List<ISymbol>();
+            foreach (var label in caseLabels)
+            {
+                // these are the labels like `MyEnum.EnumMember`
+                var memberAccessExpression = label as MemberAccessExpressionSyntax;
+                if (memberAccessExpression != null)
+                {
+                    var symbol = model.GetSymbolInfo(memberAccessExpression).Symbol;
+                    if (symbol != null)
+                    {
+                        symbols.Add(symbol);
+                    }
+                    continue;
+                }
+
+                // these are the labels like `EnumMember` (such as when using `using static Namespace.MyEnum;`)
+                var identifierName = label as IdentifierNameSyntax;
+                if (identifierName != null)
+                {
+                    var symbol = model.GetSymbolInfo(identifierName).Symbol;
+                    if (symbol != null)
+                    {
+                        symbols.Add(symbol);
+                    }
+                }
+            }
+
+            var missingLabels = new List<string>();
+            foreach (var member in enumType.GetMembers())
+            {
+                // don't create members like ".ctor"
+                if (member.IsImplicitlyDeclared)
+                {
+                    continue;
+                }
+
+                var memberExists = false;
+                foreach (var symbol in symbols)
+                {
+                    if (symbol == member)
+                    {
+                        memberExists = true;
+                        break;
+                    }
+                }
```

Comment:
> no Linq in the analyzers.
I'm not aware of that, I'll let someone else confirm this. But then, set difference doesn't have to be linq style and lazy evaluated and all that, you could simply call the extension method in imperative style like `symbols.except(members)`.